### PR TITLE
feat(plasma-core/ui): Pass onHide callback for hideToast handler

### PR DESCRIPTION
**Pull request by Yakutoc**
Добавленна возможность прокинуть callback на hideToast.  

Так же сделали overload метода `showToast`.

Так же возможность прокинуть `onHide` есть только в новой версии showToast.

Было: Аргументы передовались **inline**
  
Снимок экрана 2023-02-09 в 14 54 32

Стало: Теперь как объект.
  
Снимок экрана 2023-02-09 в 14 55 11

-----------------------------

:baby_chick: Download canary assets:

[colors--canary.319.4163108583.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.319.4163108583.xml)  
[color_metro_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.319.4163108583.swift)  
[color_metro_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.319.4163108583.kt)  
[color_metro_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.319.4163108583.ts)  
[color_sbermarket_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.319.4163108583.swift)  
[color_sbermarket_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.319.4163108583.kt)  
[color_sbermarket_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.319.4163108583.ts)  
[color_sberprime_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.319.4163108583.swift)  
[color_sberprime_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.319.4163108583.kt)  
[color_sberprime_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.319.4163108583.ts)  
[color_selgros_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.319.4163108583.swift)  
[color_selgros_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.319.4163108583.kt)  
[color_selgros_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.319.4163108583.ts)  
[color_smbusiness_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.319.4163108583.swift)  
[color_smbusiness_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.319.4163108583.kt)  
[color_smbusiness_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.319.4163108583.ts)  
[PlasmaTokensColor--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.319.4163108583.swift)  
[shadow_sbermarket_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.319.4163108583.ts)  
[typo_mage_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.319.4163108583.swift)  
[typo_mage_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.319.4163108583.kt)  
[typo_mage_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.319.4163108583.ts)  
[typo_plasma_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.319.4163108583.swift)  
[typo_plasma_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.319.4163108583.kt)  
[typo_plasma_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.319.4163108583.ts)  
[typo_ruler_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.319.4163108583.swift)  
[typo_ruler_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.319.4163108583.kt)  
[typo_ruler_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.319.4163108583.ts)  
[typo_sage_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.319.4163108583.swift)  
[typo_sage_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.319.4163108583.kt)  
[typo_sage_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.319.4163108583.ts)  
[typo_sbermarket_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.319.4163108583.swift)  
[typo_sbermarket_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.319.4163108583.kt)  
[typo_sbermarket_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.319.4163108583.ts)  
[typo_soulmate_ios-swift--canary.319.4163108583.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.319.4163108583.swift)  
[typo_soulmate_kotlin--canary.319.4163108583.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.319.4163108583.kt)  
[typo_soulmate_react-native--canary.319.4163108583.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.319.4163108583.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.138.0-canary.319.4163108583.0
  npm install @salutejs/plasma-core@1.95.0-canary.319.4163108583.0
  npm install @salutejs/plasma-icons@1.123.0-canary.319.4163108583.0
  npm install @salutejs/plasma-temple@1.132.0-canary.319.4163108583.0
  npm install @salutejs/plasma-ui@1.165.0-canary.319.4163108583.0
  npm install @salutejs/plasma-web@1.164.0-canary.319.4163108583.0
  npm install @salutejs/plasma-sb-utils@0.93.0-canary.319.4163108583.0
  # or 
  yarn add @salutejs/plasma-b2c@1.138.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-core@1.95.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-icons@1.123.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-temple@1.132.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-ui@1.165.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-web@1.164.0-canary.319.4163108583.0
  yarn add @salutejs/plasma-sb-utils@0.93.0-canary.319.4163108583.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out comparison report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

File: packages/plasma-core/src/components/Toast/Toast.types.ts [hunk 1]
```diff
@@ -9,4 +9,5 @@ export type ToastInfo = {
     timeout: number | null;
     fade?: boolean;
     role?: ToastRole;
+    onHide?: () => void;
```
Comment (neretin-trike) [inline packages/plasma-core/src/components/Toast/Toast.types.ts:12]: предлагаю сделать перегрузку showToast (по-хорошему надо было сразу), т.к. слишком много параметров в заданном порядке идут. А если ещё после этого новый добавится, например, offset, то его можно будет после колбэка вставить и тогда совсем стрёмно будет:

```ts
export interface Options {
    text: string;
    position?: ToastPosition;
    timeout?: number;
    fade?: boolean;
    contentLeft?: ReactNode;
    role?: ToastRole;
    onHide?: () => void;
}

type ContextType = ToastInfo & {
    showToast(options: Options): void;
    /** @deprecated использовать вместо этого метод с options  */
    showToast(
        text: string,
        position?: ToastPosition,
        timeout?: number,
        fade?: boolean,
        contentLeft?: ReactNode,
        role?: ToastRole,
    ): void;
    hideToast: () => void;
};
```

а так вроде и api не сломали, и даже красивенько выходит:

image

Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:12]: @neretin-trike Я не против такого улучшения =)  

@Yeti-or Как ты на такое смотришь?

Comment (Yeti-or) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:12]: Мне нравится

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

File: packages/plasma-ui/src/components/Toast/Toast.stories.tsx [hunk 2]
```diff
@@ -39,7 +41,7 @@ export const LiveDemo: Story<LiveDemoProps> = ({ role, text, position, timeout,
             </Button>
             <Button
                 onClick={() => {
-                    showToast(text, position, timeout, fade, contentLeft, role);
+                    showToast({ text, position, timeout, fade, contentLeft, role, onHide: action('onHide') });
```
Comment (neretin-trike) [inline packages/plasma-ui/src/components/Toast/Toast.stories.tsx:44]: кажется, что эти же примеры можно переделать и для b2c и web, поскольку они так же используют plasma-core

File: packages/plasma-core/src/components/Toast/Toast.types.ts [hunk 3]
```diff
@@ -2,11 +2,40 @@ import type { ReactNode } from 'react';
...
+    (
+        text: string,
+        position?: ToastPosition,
+        timeout?: number,
+        fade?: boolean,
+        contentLeft?: ReactNode,
+        role?: ToastRole,
+    ): void;
+    (options: Options): void;
+}
```
Comment (neretin-trike) [inline packages/plasma-core/src/components/Toast/Toast.types.ts:41]: а давай тут ещё немного тайп-скриптовой магии применим, это нужно, что в ToastProvider аргументы корректно разрулились без явного приведения типов:

```ts
type OldParamsType = Parameters<
    (
        text: string,
        position?: ToastPosition,
        timeout?: number,
        fade?: boolean,
        contentLeft?: ReactNode,
        role?: ToastRole,
    ) => void
>;

type CurrentParamsType = Parameters<(options: Options) => void>;

export interface ShowToast {
    /** @deprecated It is preferable to use with the "Options" implementation */
    (...args: OldParamsType): void;
    (...args: CurrentParamsType): void;
}

export type ShowToastArgsType = OldParamsType | CurrentParamsType;
```

тогда в ToastProvider будет:

```ts
    const showToastCallback: ShowToast = (...args: ShowToastArgsType) => {
        const { text, position, timeout, fade, contentLeft, role, onHide } = getShowToastParams(...args);

        setValue({
            text,
            position: position || DEFAULT_POSITION,
            timeout: timeout !== undefined ? timeout : DEFAULT_TIMEOUT,
            fade: fade !== undefined ? fade : DEFAULT_FADE,
            contentLeft,
            role,
            onHide,
        });
    };

    const showToast = useCallback(showToastCallback, []);
```

а функция getShowToastParams, просто возвращает параметры:

```ts
const getShowToastParams = (...args: ShowToastArgsType) => {
    if (typeof args[0] === 'object') {
        return args[0];
    }

    // INFO: После перехода на plasma 2.0 нужно версию с inline аргументами удалить
    const [text, position, timeout, fade, contentLeft, role] = args;
    return { text, position, timeout, fade, contentLeft, role };
};
```

Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:41]: @neretin-trike Спасибо чувак, твой вариант изящнее!  

`...args: ShowToastArgsType` <-- как раз тут я сломался, когда решал проблему, как указать что может быть **inline** и **{}**

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

File: packages/plasma-core/src/components/Toast/Toast.types.ts [hunk 4]
```diff
@@ -2,11 +2,47 @@ import type { ReactNode } from 'react';
...
+        fade?: boolean,
+        contentLeft?: ReactNode,
+        role?: ToastRole,
+    ) => void
+>;
+
+type ShowToastCallSignatureCurrent = Parameters<(options: Options) => void>;
+
+export interface ShowToast {
+    /** @deprecated It is preferable to use with the "Options" implementation */
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Toast/Toast.types.ts:43]: не понял зачем тут слово `with`

File: packages/plasma-core/src/components/Toast/ToastProvider.tsx [hunk 5]
```diff
@@ -1,40 +1,52 @@
...
 const DEFAULT_POSITION = 'bottom';
 const DEFAULT_TIMEOUT = 3000;
 const DEFAULT_FADE = true;
 
+const getShowToastCallSignature = (args: ShowToastCallSignature) => {
+    if (typeof args[0] === 'object' && 'text' in args[0]) {
+        return args[0];
+    }
+
+    // INFO: После перехода на plasma 2.0 нужно версию с inline аргументами удалить
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Toast/ToastProvider.tsx:16]: представь что мы ищем все такие места, перед выпуском 2.0 как мы их будем искать?

File: packages/plasma-core/src/components/Toast/ToastProvider.tsx [hunk 6]
```diff
@@ -1,40 +1,52 @@
...
-        },
-        [],
-    );
+    const showToastCallback: ShowToast = (...args: ShowToastCallSignature) => {
+        const { text, position, timeout, fade, contentLeft, role, onHide } = getShowToastCallSignature(args);
+
+        setValue({
+            text,
+            position: position || DEFAULT_POSITION,
+            timeout: timeout !== undefined ? timeout : DEFAULT_TIMEOUT,
```
Comment (Yeti-or) [inline packages/plasma-core/src/components/Toast/ToastProvider.tsx:31]: а почему мы position через или гардим а timeout и fade через undefined ?

File: website/plasma-ui-docs/docs/components/Toast.mdx [hunk 7]
```diff
@@ -40,34 +40,77 @@ export const App = () => (
...
 `showToast` имеет необязательный аргумент `position`, отвечающий за позиционирование подсказки.
-```jsx
+
+```ts
 type Position = 'top' | 'bottom';
 ```
 
-`showToast` имеет необязательный аргумент timeout, отвечающий за время отображения подсказки.
-Если аргумент не передается, по умолчанию подсказка отображается в течение 3 секунд. 
+`showToast` имеет необязательный аргумент ``timeout``, отвечающий за время отображения подсказки.
```
Comment (Yeti-or) [inline website/plasma-ui-docs/docs/components/Toast.mdx:83]: а чем две кавычки отличаются от одной ?
```
? => ``timeout``
? => `showToast`
```

File: packages/plasma-core/src/components/Toast/Toast.types.ts [hunk 8]
```diff
@@ -2,11 +2,47 @@ import type { ReactNode } from 'react';
...
+type ToastCommonParameters = {
     contentLeft?: ReactNode;
-    position: ToastPosition | null;
-    timeout: number | null;
     fade?: boolean;
     role?: ToastRole;
+    onHide?: () => void;
 };
+
+export type ToastInfo = ToastCommonParameters & {
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:13]: @Yeti-or Ты про какой именно тип?   

Если ты про **ToastInfo** то он используется в `ToastProvider`

`const [value, setValue] = useState<ToastInfo>({ text: null, position: null, timeout: null });`

File: packages/plasma-core/src/components/Toast/Toast.types.ts [hunk 9]
```diff
@@ -2,11 +2,47 @@ import type { ReactNode } from 'react';
...
-export type ToastInfo = {
-    text: string | null;
+
+type ToastCommonParameters = {
     contentLeft?: ReactNode;
-    position: ToastPosition | null;
-    timeout: number | null;
     fade?: boolean;
     role?: ToastRole;
+    onHide?: () => void;
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:10]: @Yeti-or Хорошее замечание - добавлю.

File: packages/plasma-b2c/src/components/Toast/Toast.stories.tsx [hunk 10]
```diff
@@ -26,14 +27,22 @@ interface LiveDemoProps extends ToastProps {
...
 export const LiveDemo: Story<LiveDemoProps> = ({ toastText, position, timeout, fade, enableContentLeft }) => {
-    const { showToast } = useToast();
+    const { showToast, hideToast } = useToast();
     const contentLeft = enableContentLeft && <IconClose size="xs" color={critical} />;
 
     return (
-        <div>
+        <div style={{ padding: '2rem 1rem' }}>
             <Button
+                style={{ zIndex: 100000000, marginRight: '1rem' }}
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-b2c/src/components/Toast/Toast.stories.tsx:36]: @Yeti-or С одной стороны ты прав. Пример не самый удачный и взят он с plasma-ui и размножен. 

Переделаю.   

> без паранжи же у нас кнопка доступна

Что это значит? )

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

File: packages/plasma-core/src/components/Toast/ToastProvider.tsx (hunk 6 above)
Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/ToastProvider.tsx:31]: @Yeti-or Такое себе оправдание, но так было )

Я бы использовал default value вместо таких guards.

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

File: packages/plasma-b2c/src/components/Toast/Toast.stories.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-b2c/src/components/Toast/Toast.stories.tsx]: я имел ввиду overlay

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

File: packages/plasma-core/src/components/Toast/ToastProvider.tsx (hunk 6 above)
Comment (Yeti-or) [reply packages/plasma-core/src/components/Toast/ToastProvider.tsx:31]: давай использовать правило бойскаутов – оставим лагерь чище чем он был до нас

Comment (Yeti-or) [reply packages/plasma-core/src/components/Toast/ToastProvider.tsx:31]: https://medium.com/swlh/11-rules-of-effective-programming-8932f2d99711

Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/ToastProvider.tsx:31]: @Yeti-or Да я только ЗА )

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

File: packages/plasma-core/src/components/Toast/Toast.types.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:20]: @Yeti-or 
Почему?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

File: packages/plasma-core/src/components/Toast/Toast.types.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (Yakutoc) [reply packages/plasma-core/src/components/Toast/Toast.types.ts:20]: Решили что оставим как есть.

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-319/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **💀 Missing report, something went wrong**

Check out current report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-319/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-319/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-319/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-319/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-319/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-319/temple-storybook/

**Review by Yeti-or — APPROVED** [review]

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**